yarp/yarp_compiler.c

```diff
@@ -1631,6 +1652,16 @@ yp_compile_node(rb_iseq_t *iseq, const yp_node_t *node, LINK_ANCHOR *const ret,
           ADD_INSNL(ret, &dummy_line_node, jump, ISEQ_COMPILE_DATA(iseq)->redo_label);
           return;
       }
+      case YP_NODE_REGULAR_EXPRESSION_NODE: {
+          if (!popped) {
+              yp_regular_expression_node_t *regular_expression_node = (yp_regular_expression_node_t *) node;
+              VALUE regex_str = parse_string(&regular_expression_node->unescaped);
```

Comment:
As a note, this will be affected by this bug: https://github.com/ruby/yarp/issues/1452
And the YP_NODE_INTERPOLATED_REGULAR_EXPRESSION_NODE case too (https://github.com/ruby/yarp/issues/1452#issuecomment-1716271367).